scm/src/scm_output.F90

```diff
@@ -392,35 +394,43 @@ subroutine output_init_diag(ncid, time_inst_id, time_diag_id, time_rad_id, hor_d
 end subroutine output_init_diag
 
 !> This subroutine appends data to the "output.nc" file.
-subroutine output_append(scm_state, physics)
+subroutine output_append(scm_state, physics, force)
 
   use scm_type_defs, only: scm_state_type, physics_type
 
   type(scm_state_type), intent(inout) :: scm_state
   type(physics_type), intent(in) :: physics
-
-  real(kind=dp), allocatable :: dummy_1D(:), dummy_2d(:,:)
+  logical, optional, intent(in) :: force
   
-  integer :: ncid, var_id, i, j
-  character(2) :: idx
-
-  allocate(dummy_1D(scm_state%n_cols), dummy_2d(scm_state%n_cols, scm_state%n_levels))
+  integer :: ncid, var_id
 
   !> \section output_append_alg Algorithm
   !! @{
-
+  if (.not. (mod(scm_state%itt, scm_state%n_itt_out)==0 .or. &
+            physics%Model%lsswr .or. physics%Model%lslwr .or. &
+            mod(scm_state%itt,physics%Model%nszero) == 0 .or. &
+            (PRESENT(force) .and. force))) then
```

Comment:
Note: This is not safe. The Fortran programming standard does NOT say that `PRESENT(force)` will be executed before testing if `force` is true, nor does it require the compiler to skip the test for `force` being true, even if the first expression evaluates to false.

You need to adjust the logic to first test for present only, and inside that if block test if force is true.

For example
```
if (present(force)) then
  force_local = force
else
  force_local = .false.
endif
```
and then use `force_local` instead of `force`.